### DISCUSSION TITLE
Connect to docker container in a remote server

### DISCUSSION THREAD

**Discussion by fabio-ivona** [spatie/ray#550] [Q&A] [answered]
Hi artisans!

I'm trying to use Ray to debug a remote system, here's my setup

Local developement machine 
↓
Remote server (ubuntu docker host)
↓
Docker Container (with host.docker.internal set up)

**note:** I can use Ray without any issue to connect to a docker container running on my local dev machine and to a non-dockerized application running on my remote server. The issue, I think, only for remote+docker setup

if I log in in my docker container and try to ping my host:
<pre>fabio@docker: $ ping host.docker.internal
PING host.docker.internal (172.17.0.1) 56(84) bytes of data.
64 bytes from host.docker.internal (172.17.0.1): icmp_seq=1 ttl=64 time=0.154 ms
</pre>

but can't connect to port 23517:
<pre>fabio@docker: $ curl host.docker.internal:23517
curl: (7) Failed to connect to host.docker.internal port 23517: Connection refused</pre>

outside the docker container (in my server shell), all works fine:
<pre>fabio@server: $ curl host.docker.internal:23517
&lt;!DOCTYPE html&gt;
&lt;html lang=&quot;en&quot;&gt;
&lt;head&gt;
&lt;meta charset=&quot;utf-8&quot;&gt;
&lt;title&gt;Error&lt;/title&gt;
&lt;/head&gt;
&lt;body&gt;
&lt;pre&gt;Cannot GET /&lt;/pre&gt;
&lt;/body&gt;
&lt;/html&gt;</pre>

some info from my server:
<pre>fabio@server: $ netstat -tulpen  | grep 23517
tcp        0      0 127.0.0.1:<font color="#C01C28"><b>23517</b></font>         0.0.0.0:*               LISTEN      1000       76498171   1916935/sshd: docke 
tcp6       0      0 ::1:<font color="#C01C28"><b>23517</b></font>               :::*                    LISTEN      1000       76498170   1916935/sshd: docke</pre>

<pre>fabio@server: $ ps -aux | grep 1916935
docker   <font color="#C01C28"><b>1916935</b></font>  0.0  0.0  12800  4408 ?        S    07:00   0:00 sshd: docker
docker   1917029  0.0  0.0   8160  2548 pts/2    S+   07:02   0:00 grep --color=auto <font color="#C01C28"><b>1916935</b></font></pre>

I think there is one issue when ray tries to forward port 23517 through ssh tunnel.

maybe it binds to localhost:23517 but can't manage to catch connection host.docker.internal:23517?

can anyone with a stronger docker knowledge help?

**Comment by fabio-ivona** [comment, marked as answer]
I found the issue:

my sshd config was forcing to bind only internal connection (see **127.0.0.1:23517**)

I solved by enabling this setting in /etc/ssh/sshd_config:

`GatewayPorts yes`

@freekmurze maybe this should be added to Ray docs? In linux systems the default is set to `no`

**Reply by freekmurze** [reply]
@fabio-ivona feel free to PR this to our docs (there's a little edit button on top of every page)

**Reply by fabio-ivona** [reply]
done!
